modified getEmailFromResource function to get resourceOwner details f…

laky241 opened the pull request:
Fixes #2330 


Modified **getEmailFromOAuth** Function

updated to call a function, **getResourceFromToken** which extracts the required resource owner ID.

Updated the OIDC scope to include **profile** which claims like **upn**

shaheemazmalmmd commented:
@laky241 : can you explain 
* why this changes are made ?
* how to test these changes ?

laky241 commented:
@shaheemazmalmmd 

I made changes following the solution provided in the issue and were tested successfully on Fossology Version (85cf465)

getEmailFromResource, which wasn't flexible for all OIDC providers. to fetch details 

introduced a new function, getResourceFromToken, to directly decode the OIDC token to extract the resource owner details.

Updated scope from ['email openid'] to ['openid profile email'] to include additional claims like email and username.

please guide if i have missed anything

shaheemazmalmmd commented:
> @shaheemazmalmmd
> 
> I made changes following the solution provided in the issue and were tested successfully on Fossology Version ([85cf465](https://github.com/fossology/fossology/commit/85cf465677b61ccb2de7f022cfc3925114052af1))
> 
  > getEmailFromResource, which wasn't flexible for all OIDC providers. to fetch details
  > 
  > introduced a new function, getResourceFromToken, to directly decode the OIDC token to extract the resource owner details.
> 
> Updated scope from ['email openid'] to ['openid profile email'] to include additional claims like email and username.
> 
> please guide if i have missed anything

@laky241 : the existing ['email openid'] fetch's the required information that is useful. why additional information is needed when it is not stored/used.

getResourceFromToken i was not able to see any code related to this function. 

also use of validateAccessToken has silently been removed.
